src/redisearch_rs/tracing_redismodule/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+//! A subscriber for the Rust `tracing` ecosystem that emits traces and logs to the redismodule logging system.
+//!
+//! # Configuring Logging Output
+//!
+//! Logging output can be configured by setting the `RUST_LOG` environment variable to a _filter_.
+//! A filter consists of one or more comma-separated directives which match on `Span`s and `Event`s.
+//! Each directive may have a corresponding maximum verbosity [`level`] which enables (e.g., _selects for_)
+//! spans and events that match. Like `log`, `tracing` considers less exclusive levels (like `trace` or `info`)
+//! to be more verbose than more exclusive levels (like `error` or `warn`).
+//!
+//! The directive syntax is similar to that of [`env_logger`]'s. At a high level, the syntax for directives
+//! consists of several parts:
+//!
+//! ```text
+//! target[span{field=value}]=level
+//! ```
+//!
+//! Each component (`target`, `span`, `field`, `value`, and `level`) will be covered in turn.
+//!
+//! - `target` matches the event or span's target. In general, this is the module path and/or crate name.
+//!   Examples of targets `h2`, `tokio::net`, or `tide::server`. For more information on targets,
```

Comment:
Can't this instead link to the `env_logger` documentation for setting up the filter?

Reply:
Its not actually `env_logger` syntax because this includes tracing specific extensions, the tracing docs aren't actually that helpful for ppl not already familiar with with logging so I chose to copy and adapt the docs here. But no hard preference either way, I'm fine with removing as well

Reply:
Sorry, I meant just linking to this instead after the small example -> https://docs.rs/tracing-subscriber/0.3.20/tracing_subscriber/filter/struct.EnvFilter.html